src/Tasks/FileIO/WriteLinesToFile.cs

```diff
@@ -122,6 +122,10 @@ public override bool Execute()
                     }
                     else
                     {
+                        if (WriteOnlyWhenDifferent)
+                        {
+                            Log.LogWarningWithCodeFromResources("WriteLinesToFile.UnusedWriteOnlyWhenDifferent", File.ItemSpec);
```

Comment:
Ideally I think this would have been an error from the introduction of `WriteOnlyWhenDifferent`. Since it wasn't, we shouldn't make it an error now, but in our codebase that generally also includes new _warnings_ since many builds have WarnAsError on.

So I'm not sure this is the right thing to do now. I see two main options:

1. We could emit the warning behind changewave 17.6, in the hope that basically no one is hitting this so we can turn it on breaking no one and then be in a fairly good state.
2. We could emit it as a (high-priority?) _message_ instead, which is always nonbreaking.

I think I'm leaning toward option 1, at least for 17.6-preview.2. @baronfel thoughts appreciated.

Reply:
Applied the wave check in the latest iteration.